keps/sig-apps/0032-portable-service-definitions.md

```diff
@@ -0,0 +1,157 @@
+# KEP: Portable Service Definitions
+
+---
+kep-number: 31
+title: Portable Service Definitions
+authors:
+  - "@mattfarina"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-service-catalog
+reviewers:
+  - "@carolynvs"
+  - "@kibbles-n-bytes"
+  - "@duglin"
+  - "@jboyd01"
+  - "@prydonius"
+  - "@kow3ns"
+approvers:
+  - "@mattfarina"
+  - "@prydonius"
+  - "@kow3ns"
+editor: TBD
+creation-date: 2018-11-13
+last-updated: 2018-11-19
+status: provisional
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Portable Service Definitions
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a KEP and for highlighting any additional information provided beyond the standard KEP template.
+[Tools for generating][] a table of contents from markdown are available.
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+The goal of this feature is to enable an application to be deployed into multiple environments while relying on external services that are not part of the application and using the same objects in all environments. That includes service instances may or may not be managed by Kubernetes. For example, take a WordPress application that relies on MySQL. If the application is running in GKE or AKS it may leverage a Google or Azure MySQL as a Service. If it is running in Minikube or in a bare metal cluster it may use MySQL managed by the Kubernetes cluster. In all of these cases, the same resource is declared asking for the service (e.g., MySQL) and the credentials to use the service are returned in a common manner (e.g., a secret with the same type and schema). This enables portability from one environment to another because working with services in all environments leverages the same Kubernetes objects with the same schemas.
```

Comment:
That includes service instances **that** may or may not be managed by Kubernetes